app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,6 @@
+class RecipesController < ApplicationController
+  def show 
+    @recipe = Recipe.find(params[:id])
+    @total_cost = @recipe.ingredients.sum(:cost)
```

Comment:
Similar to your `IngredientsController`, doing `.sum` here is correct AR, but it is doing a calculation on the database. As such, it should be placed in a model method and tested.

Reply:
Also regarding model tests, a little spoiler/hint for you in future ICs.... if you ever finish an IC and realize that you didn't make any new model tests, that should be an alarm that sounds in your mind to go back and double-check for MVC issues!